### PR TITLE
feature/mx-1658 fix wikidata 403 bug

### PR DESCRIPTION
**Pull request by mr-kamran-ali**
### Added

- HTTP connector backoff for 10 retries on 403 from server
- `rki/mex` user agent is sent with query requests via wikidata connector

### Changes

- update wikidata search organization request query, with optional language parameter
  wikidata query search can be enhanced by specifying the language.
  EN is the default language.

### CODE REVIEW

**Comment by cutoffthetop** [comment]
@mr-kamran-ali can retrying really fix a 403 error? 🤔

**Comment by mr-kamran-ali** [comment]
> @mr-kamran-ali can retrying really fix a 403 error? 🤔

It probably doesn't, issue was fixed by the `User-Agent` headers, I took this step to give 10 backoff attempts because it was still working on some requests and failing on others. Also with this extractor will fail bit more elegantly.

File: mex/common/wikidata/connector.py [hunk 1]
```diff
@@ -77,6 +78,7 @@ def get_wikidata_item_details_by_id(self, item_id: str) -> dict[str, str]:
             ),
             "formatversion": "2",
         }
+        headers = {"User-Agent": "rki/mex", "Api-User-Agent": "rki/mex"}
```
Comment (erichesse) [inline mex/common/wikidata/connector.py:81]: I vote for DRY here. Would be better to put this into the mex.common.settings

File: mex/common/wikidata/extract.py [hunk 2]
```diff
@@ -11,31 +11,41 @@
...
+        f'bd:serviceParam mwapi:search "{item_label}" . '
+        f'bd:serviceParam mwapi:language "{lang}" . '
+        "?item wikibase:apiOutputItem mwapi:item . "
+        "?num wikibase:apiOrdinal true . "
+        "} "
+        "?item (wdt:P31/wdt:P8225*/wdt:P279*) wd:Q43229. "
+        'SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en,de". } '  # noqa: E501
+        "} "
+        "ORDER BY ASC(?num) "
+        "LIMIT 1 "
```
Comment (erichesse) [inline mex/common/wikidata/extract.py:43]: Could you explain what the WHERE clause is doing? I don't understand it

**Review by erichesse — COMMENTED** [review]
The tests are failing for me locally. for `pdm run pytest tests/wikidata` I get:
```
E               requests.exceptions.ProxyError: HTTPSConnectionPool(host='wikidata', port=443): Max retries exceeded with url: /https://wikidata/?format=json&action=wbgetentities (Caused by ProxyError('Unable to connect to proxy', OSError('Tunnel connection failed: 503 Service Unavailable')))

.venv\Lib\site-packages\requests\adapters.py:694: ProxyError
```
at `tests\wikidata\test_extract.py:521`
and `tests\wikidata\test_extract.py:26`
and `tests\wikidata\test_extract.py:34`
and `tests\wikidata\test_extract.py:55`
and `tests\wikidata\test_connector.py:100`
and `tests\wikidata\test_extract.py:66`
and `tests\wikidata\test_connector.py:59`


Also maybe outside of the scope of this ticket. But shouldn't we integrate wikidata in a way into the extractors, that they don't fail, if wikidata doesn't work. wikidata seems to be too unreliable to critically rely on it.

File: mex/common/wikidata/connector.py (hunk 1 above)
Comment (mr-kamran-ali) [reply mex/common/wikidata/connector.py]: will something like this work? 5d4e668b8103bed123edb109824b19a2899949a9

entire dict can vary from service to service, so it wont be reuseable everywhere if we move it to settings.

File: mex/common/wikidata/extract.py (hunk 2 above)
Comment (mr-kamran-ali) [reply mex/common/wikidata/extract.py:43]: its part of the MediaWiki API Query Service. Its faster and has better caching than the older query we were using here. I am not sure how much technically superior it is.

WHERE clause is combining some parameters and requesting internal MediaWiki API, specifying:
- Which action to use: `EntitySearch`
- Which endpoint to use: `wikidata.org` they have wikipedia as well. 
- What to search: `item_label`
- Which language to search in: `lang`
- What do we expect it to return: `item` an organization or person is an item in wikidata
- apiOrdinal is used for ordering the results to get the first best match

you can read more about it here: https://www.mediawiki.org/wiki/Wikidata_Query_Service/User_Manual/MWAPI

**Review by erichesse — APPROVED** [review]

**Pull request merged**